site/app/views/forum/ForumThreadView.php

```diff
@@ -182,81 +182,94 @@ function openFile(directory, file, path ){
-
-			<a class="btn btn-primary" style="position:relative;top:3px;left:5px;" title="Create thread" onclick="resetScrollPosition();" href="{$this->core->buildUrl(array('component' => 'forum', 'page' => 'create_thread'))}"><i class="fa fa-plus-circle"></i> Create Thread</a>
+		<a class="btn btn-primary" style="position:relative;top:3px;left:5px;" title="Create thread" onclick="resetScrollPosition();" href="{$this->core->buildUrl(array('component' => 'forum', 'page' => 'create_thread'))}"><i class="fa fa-plus-circle"></i> Create Thread</a>
 HTML;
-
-		if($this->core->getUser()->getGroup() <= 2){
-			$return .= <<<HTML
+	if($this->core->getUser()->getGroup() <= 2){
+		
+		$return .= <<<HTML
 			<a class="btn btn-primary" style="margin-left:10px;position:relative;top:3px;right:5px;display:inline-block;" title="Show Stats" onclick="resetScrollPosition();" href="{$this->core->buildUrl(array('component' => 'forum', 'page' => 'show_stats'))}">Stats</a>
 HTML;
-		}
-				$categories = $this->core->getQueries()->getCategories();
-				$onChange = '';
-				if($thread_count > 0) {
-					$onChange = <<<HTML
-						onchange="modifyThreadList({$currentThread}, {$currentCategoryId[0]["category_id"]});"
+	}
+	$categories = $this->core->getQueries()->getCategories();
+	$onChange = '';
+	if($thread_count > 0) {
+		$onChange = <<<HTML
+		onchange="modifyThreadList({$currentThread}, {$currentCategoryId[0]["category_id"]});"
 HTML;
-				}
-				$return .= <<<HTML
-				<div style="display:inline-block;position:relative;top:3px;margin-left:5px;" id="category_wrapper">
-				<label for="thread_category">Category:</label>
-			  	<select id="thread_category" name="thread_category" class="form-control" {$onChange}>
-			  	<option value="" selected>None</option>
+	}
+	$return .= <<<HTML
+		<div style="display:inline-block;position:relative;top:3px;margin-left:5px;" id="category_wrapper">
+		<label for="thread_category">Category:</label>
+	  	<select id="thread_category" name="thread_category" class="form-control" {$onChange}>
+	  	<option value="" selected>None</option>
 HTML;
-			    for($i = 0; $i < count($categories); $i++){
-			    	$return .= <<<HTML
-			    		<option value="{$categories[$i]['category_id']}">{$categories[$i]['category_desc']}</option>
+	    for($i = 0; $i < count($categories); $i++){
+	    	$return .= <<<HTML
+	    		<option value="{$categories[$i]['category_id']}">{$categories[$i]['category_desc']}</option>
 HTML;
-			    } 
+	    } 
 
-$return .= <<<HTML
+	$return .= <<<HTML
 			</select>
 			</div>
 			<button class="btn btn-primary" style="float:right;position:relative;top:3px;right:5px;display:inline-block;" title="Display search bar" onclick="this.style.display='none'; document.getElementById('search_block').style.display = 'inline-block'; document.getElementById('search_content').focus();"><i class="fa fa-search"></i> Search</button>
+HTML;
+	if($this->core->getUser()->getGroup() <= 2){
```

Comment:
We are checking user authorization in UI but not in the backend therefore even student can add `option` parameter in url directly to change view.
```php
$option = $_REQUEST['option'] ?? 'tree';
```

https://github.com/Submitty/Submitty/pull/1916/files#diff-82b25ad2bdc78a61032e380fe360d299R295

Reply:
Good catch, I believe that this can be fixed by simply changing the above expression to a ternary expression that also includes a check for the group being <= 2.